pytest_fixtures/content_hosts.py

```diff
@@ -26,52 +38,113 @@ def rhel_contenthost(request):
         yield host
 
 
-@pytest.fixture
-def rhel7_contenthost():
-    """A function-level fixture that provides a content host object based on the rhel7 nick"""
-    with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as host:
+@pytest.fixture(
+    params=[{'workflow': settings.content_host.deploy_workflow, 'rhel': '7'}],
+    ids=[settings.content_host.deploy_workflow],
+)
+def rhel7_contenthost(host_conf):
+    """A function-level fixture that provides a rhel7 content host object"""
+    with VMBroker(
+        workflow=host_conf['workflow'],
+        rhel_version=host_conf['rhel'],
+        memory=host_conf['memory'],
+        cores=host_conf['cores'],
+        host_classes={'host': ContentHost},
+    ) as host:
         yield host
 
 
 @pytest.fixture
 def rhel7_contenthost_fips():
     """A function-level fixture that provides a content host object based on the rhel7_fips nick"""
-    with VMBroker(nick='rhel7_fips', host_classes={'host': ContentHost}) as host:
+    rhel7 = settings.broker.clients.rhel7
+    with VMBroker(
+        workflow='deploy-base-rhel-fips',
+        rhel_version=rhel7.release,
+        memory=rhel7.memory,
+        cores=rhel7.cores,
+        host_classes={'host': ContentHost},
+    ) as host:
         yield host
 
 
-@pytest.fixture(scope="module")
-def rhel7_contenthost_module():
+@pytest.fixture(
+    scope='module',
+    params=[(settings.content_host.deploy_workflow, '7')],
```

Comment:
Bad idea. These can change on reconfigure of AT and DEFINITELY change whenever we re-deploy... the WF order for configuration changes (like if I re-order them in YAML and redeploy).

Reply:
its not, 7 is the rhel major version.